app.js

```diff
@@ -48,10 +37,38 @@ angular.module('PathOfDamage', [])
         }
       }
     }
-    $scope.updateHits(true);
-    if (!skipStringify) {
-      $scope.stringifyUrlData();
+    if (!skipUpdates) {
+      $scope.updateHits();
+    }
+  };
+
+  $scope.updateHits = function (skipSerialization) {
+    for (var i = 0; i < $scope.hits.length - 1; i++) {
+      $scope.hits[i] = calcDamage($scope.hits[i].hit);
+    }
+    if (!skipSerialization) {
+      serializeData();
+    }
+  };
+
+  $scope.updateHitTable = function (index) {
```

Comment:
This really only updates one or two rows, not the whole table, right? I mean, I know it sometimes adds a row, but its more like updating a row or two than updating the entire table. It seems unintuitive that a table updater would take an index number.

 I don't really know what a better name would be. UpdateHitRow seems clunky.